ONSdigital/dp-api-tests
# Update filter API tests to include new download structure

### What

Update filter API tests to include new download structure

Also extended tests to include new validation rules based on whether
the public or private download link is available and whether the
instance is published.

### How to review

Check changes make logical sense and filterAPI tests pass.

Run filterAPI tests by doing the following:
1) Run `dp-auth-api-stub` on cmd-develop
2) Run `dp-dataset-api` on `cmd-develop`
3) Run `dp-filter-api` on `feature/private-public-links`

The e2e tests will fail as it is dependent on existing work

### Who can review

Describe who worked on the changes, so that other people can review.
